nucypher/cli/characters/ursula.py

```diff
@@ -333,151 +318,134 @@ def ursula(click_config,
-            click.echo(row)
-
-        click.echo("Select ethereum address")
-        account_selection = click.prompt("Enter 0-{}".format(len(ursula_config.accounts)), type=click.INT)
-        address = click_config.accounts[account_selection]
-
-    if action == 'list':
-        live_stakes = ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address)
-        for index, stake_info in enumerate(live_stakes):
-            row = '{} | {}'.format(index, stake_info)
-            click.echo(row)
-
-    elif action == 'init':
-        click.confirm("Stage a new stake?", abort=True)
-
-        live_stakes = ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address)
-        if len(live_stakes) > 0:
-            raise RuntimeError("There is an existing stake for {}".format(checksum_address))
-
-        # Value
-        balance = ursula_config.miner_agent.token_agent.get_balance(address=checksum_address)
-        click.echo("Current balance: {}".format(balance))
-        value = click.prompt("Enter stake value", type=click.INT)
+    elif action == 'stake':
+
+        # List only
+        if list_:
+            live_stakes = list(URSULA.miner_agent.get_all_stakes(miner_address=URSULA.checksum_public_address))
+            if not live_stakes:
+                click.echo(f"There are no existing stakes for {URSULA.checksum_public_address}")
+            painting.paint_stakes(stakes=live_stakes)
+            return
+
+        # Confirm new stake init
+        if not force:
+            click.confirm("Stage a new stake?", abort=True)
+
+        # Validate balance
+        balance = URSULA.token_agent.get_balance(address=URSULA.checksum_public_address)
+        if balance == 0:
+            click.secho(f"{ursula.checksum_public_address} has 0 NU.")
+            raise click.Abort
+        if not quiet:
+            click.echo(f"Current balance: {balance}")
+
+        # Gather stake value
+        if not value:
+            min_stake_nu = int(URSULA.blockchain.interface.w3.fromWei(MIN_ALLOWED_LOCKED, 'ether'))
+            value = click.prompt(f"Enter stake value in NU", type=click.INT, default=min_stake_nu)
+        stake_wei = URSULA.blockchain.interface.w3.toWei(value, 'ether')
+        stake_nu = value
 
         # Duration
-        message = "Minimum duration: {} | Maximum Duration: {}".format(MIN_LOCKED_PERIODS, MAX_MINTING_PERIODS)
-        click.echo(message)
-        duration = click.prompt("Enter stake duration in periods (1 Period = 24 Hours)", type=click.INT)
-
-        start_period = ursula_config.miner_agent.get_current_period()
+        if not quiet:
+            message = "Minimum duration: {} | Maximum Duration: {}".format(MIN_LOCKED_PERIODS, MAX_MINTING_PERIODS)
+            click.echo(message)
+        if not duration:
+            duration = click.prompt("Enter stake duration in periods (1 Period = 24 Hours)", type=click.INT)
+        start_period = URSULA.miner_agent.get_current_period()
```

Comment:
Shouldn't min and max duration be enforced here? Currently, it allows me to do this:
```
Minimum duration: 30 | Maximum Duration: 365
Enter stake duration in periods (1 Period = 24 Hours): 29

============================== STAGED STAKE ==============================

(Ursula)⇀Aquamarine Scorpius Coral Queen↽ (0xF59CFb62c66be1dBe0B7b66f2108cb3bFE91667F)
~ Value      -> 15000 NU (15000000000000000000000 NU-wei) 
~ Duration   -> 29 Days (29 Periods)
~ Enactment  -> 2019-03-13 12:16:09.982821+00:00 (period #17968)
~ Expiration -> 2019-04-11 12:16:09.983282+00:00 (period #17997)
    
=========================================================================

```

Or even this!!

```
Enter stake duration in periods (1 Period = 24 Hours): -1

============================== STAGED STAKE ==============================

(Ursula)⇀Aquamarine Scorpius Coral Queen↽ (0xF59CFb62c66be1dBe0B7b66f2108cb3bFE91667F)
~ Value      -> 15000 NU (15000000000000000000000 NU-wei) 
~ Duration   -> -1 Days (-1 Periods)
~ Enactment  -> 2019-03-13 12:30:29.387475+00:00 (period #17968)
~ Expiration -> 2019-03-12 12:30:29.387703+00:00 (period #17967)
    
=========================================================================
```

Reply:
Haha, yes indeed, needs better validation.

Reply:
:heavy_check_mark:

Reply:
Hmm, that's interesting btw. What happens if you stake for negative time?

Reply:
It will fall validation, or if you bypass all of it, the transaction will fail.

Reply:
Turns out even though the expected behavior is to fail, this particular check does indeed allow a successful transaction with a negative value for periods. @szotov